src/cpp/core/Error.cpp

```diff
@@ -155,6 +156,20 @@ bool Error::isError() const
       return false ;
 }
 
+bool Error::isShutdownError(const boost::system::error_code& ec)
```

Comment:
This is also defined here: https://github.com/rstudio/rstudio/blob/master/src/cpp/session/http/SessionHttpConnectionListenerImpl.hpp#L55-L70 (that's likely where you got the implementation?). Let's move this to core/include/core/http/SocketUtils.hpp